client-cm/src/main/java/com/sequenceiq/cloudbreak/cm/client/tracing/StackTraceBasedCmApiNameExtractor.java

```diff
@@ -0,0 +1,34 @@
+package com.sequenceiq.cloudbreak.cm.client.tracing;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StackTraceBasedCmApiNameExtractor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackTraceBasedCmApiNameExtractor.class);
+
+    public Optional<String> getCmApiName(StackTraceElement[] stackTrace) {
+        List<StackTraceElement> list = new ArrayList<>();
+        Collections.addAll(list, stackTrace);
+        Optional<StackTraceElement> first = list.stream()
+                .filter(e -> e.getClassName().startsWith("com.cloudera.api.swagger"))
+                .filter(e -> !e.getClassName().startsWith("com.cloudera.api.swagger.client"))
+                .findFirst();
+
```

Comment:
this empty line indicates this method should be splitted into 2 separate at least

Reply:
I will rewrite this to use stack walker